### ISSUE TITLE
[OOM error] res = learn.get_preds(DatasetType.Valid)

### ISSUE THREAD

**Issue by KeremTurgutlu** [fastai/fastai#2337]
https://github.com/fastai/fastai/blob/5300b7633b9011c5c9eaab0544b1e3368a279ec4/fastai/basic_train.py#L29

Is there a specific reason why `.detach()` is used for ` yb[0].detach()` instead of `to_detach(yb[0])`. Keeping `yb[0]`s in gpu accumulates memory thus throws OOM error when `get_preds` is called in big validation sets.

if `.detach()` is not intended for a special purpose instead `to_detach()` might be used to solve this issue.

**Comment by sgugger** [comment]
I don't see a reason why we're not using `to_detach`, will fix later this morning.

**Comment by KeremTurgutlu** [comment]
Thanks :)
